frontend/cypress/integration/common/graph_display.ts

```diff
@@ -0,0 +1,365 @@
+  cy.on('uncaught:exception', (err, runnable, promise) => {
+    // when the exception originated from an unhandled promise
+    // rejection, the promise is provided as a third argument
+    // you can turn off failing the test in this case
+    if (promise) {
+      return false;
+    }
+    // we still want to ensure there are no other unexpected
+    // errors, so we let them fail the test
+  });
+});
+
+When('user graphs {string} namespaces', namespaces => {
+  // Forcing "Pause" to not cause unhandled promises from the browser when cypress is testing
+  cy.visit(url + `/graph/namespaces?refresh=0&namespaces=${namespaces}`);
+});
+
+When('user opens display menu', () => {
+  cy.get('button#display-settings').click();
+});
+
+When('user enables {string} {string} edge labels', (radio, edgeLabel) => {
+  cy.get('button#display-settings').get(`input#${edgeLabel}`).check();
+  cy.get(`input#${radio}`).check();
+});
+
+When('user {string} {string} edge labels', (action, edgeLabel) => {
+  if (action === 'enables') {
+    cy.get('button#display-settings').get(`input#${edgeLabel}`).check();
+  } else {
+    cy.get('button#display-settings').get(`input#${edgeLabel}`).uncheck();
+  }
+});
+
+When('user {string} {string} option', (action, option: string) => {
+  let id: string;
+  switch (option.toLowerCase()) {
+    case 'cluster boxes':
+      option = 'boxByCluster';
+      break;
+    case 'idle edges':
+      option = 'filterIdleEdges';
+      break;
+    case 'idle nodes':
+      option = 'filterIdleNodes';
+      break;
+    case 'missing sidecars':
+      option = 'filterSidecars';
+      break;
+    case 'namespace boxes':
+      option = 'boxByNamespace';
+      break;
+    case 'rank':
+      option = 'rank';
+      break;
+    case 'service nodes':
+      option = 'filterServiceNodes';
+      break;
+    case 'security':
+      option = 'filterSecurity';
+      break;
+    case 'traffic animation':
+      option = 'filterTrafficAnimation';
+      break;
+    case 'virtual services':
+      option = 'filterVS';
+      break;
+    default:
+      option = 'xxx';
+  }
+
+  if (action === 'enables') {
+    cy.get('button#display-settings').get(`input#${option}`).check();
+    if (option === 'rank') {
+      cy.get(`input#inboundEdges`).check();
+    }
+  } else {
+    cy.get('button#display-settings').get(`input#${option}`).uncheck();
+  }
+});
+
+///////////////////
+
+Then(`user sees no namespace selected`, () => {
+  cy.get('div#empty-graph-no-namespace').should('be.visible');
+});
+
+Then(`user sees the {string} namespace`, ns => {
+  cy.get('div#summary-panel-graph').find('div#summary-panel-graph-heading').find(`span#ns-${ns}`).should('be.visible');
+});
+
+Then('the display menu opens', () => {
+  cy.get('button#display-settings').invoke('attr', 'aria-expanded').should('eq', 'true');
+});
+
+Then('the display menu has default settings', () => {
+  let input = cy.get('button#display-settings').get(`input#responseTime`);
```

Comment:
Is the `.within` approach any different than using `.find`?

Also, why is is frowned upon to do the assigment?  What's the recommendation, to always chain?  Or, as stated below, perhaps look at `data-test`?

Reply:
> Is the `.within` approach any different than using `.find`?

Both of these will be scoped to the resource you are chaining off of unlike when chaining `.get`.

> Also, why is is frowned upon to do the assigment? What's the recommendation, to always chain? Or, as stated below, perhaps look at `data-test`?

Because the cypress commands get executed asynchronously and the api is supposed to work in a similar manner to promises: https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Return-Values. Since you are just doing assertions on the element it may not matter as much but if you had some kind of interaction like `input.click()` then there might be some weirdness.

>  What's the recommendation, to always chain?

Yes chaining is the way to go. You can use `and` to chain assertions.